### PR TITLE
add upgrade note for 4.0 to changelog

### PR DESCRIPTION
**Pull request by minrk**
I also opened #4425 because I think the changelog isn't really the best place for upgrading instructions, but that's a bigger task.

### CODE REVIEW

**Comment by minrk** [comment]
linkcheck should be fixed by #4427

File: docs/source/reference/changelog.md [hunk 1]
```diff
@@ -13,6 +13,14 @@ command line for details.
 ### 4.0.0 - 2023-04-20
 
 4.0 is a major release, but a small one.
+
+:::{note}
+**Upgrade note**
```
Comment (consideRatio) [inline docs/source/reference/changelog.md:18]:
```suggestion
:::{admonition} Upgrade note
```

Like this, we get a note section with the title "Upgrade note" as compared to having it titled "Note" followed by the bold text "Upgrade note" inside it.

Comment (consideRatio) [reply docs/source/reference/changelog.md:18]: image

Compared to

image

Comment (consideRatio) [reply docs/source/reference/changelog.md:18]: Either seems fine, merge directly if you want

**Review by consideRatio — APPROVED** [review]

**Pull request merged**